docs/guide/writing-migrations.md

```diff
@@ -0,0 +1,262 @@
+
+The second is `limit` and is optional. It allows you to pass an integer that specifies how many documents to process in parallel. If no integer is passed as the second argument it is 1 by default.
+
+The third criteria is the `iterator` function that should be performed on every document found. It receives the document as an argument. You can use most MongoDB methods here, but typically it uses the [`updateOne` method](https://www.mongodb.com/docs/drivers/csharp/current/usage-examples/updateOne/) to modify the document being passed to the iterator.
+
+Here is an example migration that adds a `featured` boolean property to all `article` pieces, defaulting to `false`:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+   async init(self) {
+    self.apos.migration.add('change-featured-flag', self.changeFeaturedFlag);
+  }
+  methods(self) {
+    return {
+      async changeFeaturedFlag(self) {
+        return self.apos.migration.eachDoc({
+          type: 'article'
+        }, async (doc) => {
+          if (doc.featured === undefined) {
+            await self.apos.doc.db.updateOne({
+              _id: doc._id
+            }, {
+              $set: { featured: false }
+            });
+          }
+        });
+      }
+    }
+  }
+};
+```
+<template v-slot:caption>
+  /modules/article/index.js
+</template>
+
+</AposCodeBlock>
+
+In this example:
+
+- The `eachDoc` method iterates over all documents, finding those with the `type` of `article`.
+- For each found document, we check if the `featured` property is missing.
+- We use the shorthand `self.apos.doc.db` to access the `aposDocs` collection of our database.
+- The [`updateOne`](https://www.mongodb.com/docs/drivers/node/current/fundamentals/crud/write-operations/modify/) helper operation allows us to modify the document by passing in the document `_id`.
+- Finally, we use the [`$set` operator](https://www.mongodb.com/docs/manual/reference/operator/update/set/) to add the `featured` property without modifying any other fields.
+
+::: info
+Note: In this example we are checking if the `featured` property is missing before using `$set`. This will prevent overwriting any existing values in the database. This might not be the behavior you intend. You might want all `featured` schema fields to have a value of false as a default. In this case, just skip the check and the `$set` operation will either create the featured field or change the value to `false` if it already exists.
+:::
+
+## Removing a Property from Existing Documents
+
+If you need to remove a property, you can use `$unset`. Here’s an example that removes a `temporaryNote` property from all `default` page types:
```

Comment:
I agree we should cover how to unset, but please add an info note stating that it is often wiser to leave the data you think you "don't need anymore" in the database and simply remove it from the schema, at least at first, to make absolutely sure you won't regret deleting information.

Reply:
Added.